lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -164,5 +164,5 @@ def test_query_counts_uncached(self, store_type_tuple, with_storage_backing):
             self._get_blocks(
                 course,
                 expected_mongo_queries,
-                expected_sql_queries=14 if with_storage_backing else 6,
+                expected_sql_queries=28 if with_storage_backing else 20,
```

Comment:
That's a big jump in queries. Is this adding a separate select for every block?

Reply:
If so, is there a clean way to collapse that to a single query? We know up front what course and user it is. Do we have the appropriate fields to do a scan on (user, course), or to do a "select in" with all the appropriate block_ids?

Reply:
Done reduced queries.